queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,61 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Simple directives
+(directive name: _ @keyword)
+
+
+;; Rule imports
+(rule_import
+  "use" @include
+  "rule" @include
+  "from" @include
+)
+
+
+;; Rule inheritance
+(rule_inheritance
+  "use" @keyword
+  "rule" @keyword
+  "from"? @include
+)
```

Comment:
Why are these different?

Reply:
Thank you for taking a look!

You're right, this is not entirely consistent. The difference is that a rule import takes a rule from another workflow file, whereas an inheritance takes a rule from the same file, but modifies some of its parameters. However, it's also possible to do a modifying rule import, which imports a rule and immediately modifies it. The `use rule (from)` syntax is used for both.

I should probably use `@type` for rule names though and adjust the `as` for imports.

![smk_py](https://user-images.githubusercontent.com/8322329/218320426-a8dd6760-f7f7-48a6-956b-eb28e2888a90.jpg)

Reply:
> However, it's also possible to do a modifying rule import, which imports a rule and immediately modifies it. 

That's still an import. :)

Reply:
> > However, it's also possible to do a modifying rule import, which imports a rule and immediately modifies it.
> 
> That's still an import. :)

Fair enough. I've updated the parser and the queries to reflect that.
Not entirely sure what to do with the final optional `with` for a modifying import. I've left it as `@include` for now,  what do you think?
![smk](https://user-images.githubusercontent.com/8322329/218326508-061c786a-6ab2-401a-94e3-f41104f97380.jpg)